pyccel/codegen/printing/fcode.py

```diff
@@ -1089,12 +1291,53 @@ def _print_Assign(self, expr):
         if isinstance(rhs, (PythonRange, Product)):
             return ''
 
-        if isinstance(rhs, (NumpyArray, NumpyLinspace, NumpyDiag, NumpyCross,\
-						NumpyWhere, PyccelArraySize)):
+        if isinstance(rhs, (PythonLen, NumpyRandint)):
+            rhs_code = self._print(rhs)
+            return '{0} = {1}\n'.format(lhs_code, rhs_code)
+
+        if isinstance(rhs, PythonInt):
+            rhs = expr.rhs.fprint(self._print)
+            return '{0} = {1}\n'.format(lhs_code,rhs)
+        
+        if isinstance(rhs, (NumpyComplex, NumpyReal, NumpyArray, NumpyLinspace)):
+            rhs = self._print(rhs)
+            return '{0} = {1}\n'.format(lhs_code,rhs)
+        
+        if isinstance(rhs, (NumpyCross, NumpyDiag, NumpyWhere)):
+            return self._print(rhs, expr.lhs)
+        
+        if isinstance(rhs, PyccelArraySize):
             return rhs.fprint(self._print, expr.lhs) + '\n'
 
-        if isinstance(rhs, (NumpyFull, NumpyFullLike, NumpyEmptyLike,\
-						NumpyZerosLike, NumpyOnesLike, NumpyRand)):
+
+        if isinstance(rhs, NumpyRand):
+
+            stack_array = False
+            if self._current_function:
+                name = self._current_function
+                func = self.get_function(name)
+                lhs_name = expr.lhs.name
+                vars_dict = {i.name: i for i in func.local_vars}
+                if lhs_name in vars_dict:
+                    stack_array = vars_dict[lhs_name].is_stack_array
```

Comment:
This doesn't look like it should be in the printing. Was it there before? @saidctb do you know what this does?

Reply:
this should be removed, it handles stack arrays , so that we don't allocate them

Reply:
it's probably not needed anymore

Reply:
Given that allocation is handled with the `Allocate` node that sounds right to me. @mostafaMamoni this code, and the identical code below it should be removed